test.py

```diff
@@ -1,30 +1,53 @@
+from crawler import config
 from crawler.rss import RSSCrawler
 from crawler.github import GithubCrawler
-from crawler.config import RSSConfig as conf
-import json, sys
+from crawler.config import RSSConfig as rss_conf
+import unittest
+import requests
 
-def test_rss():
-    # RSS test
-    # TODO prevent from unappropriate input
-    newspaper = input("[+] Newspaper name: ")
-    rss_result = RSSCrawler(newspaper)
-    print(json.dumps(rss_result, indent=4))
+class TestRSSCrawler(unittest.TestCase):
+    """
+    Testcase for RSSCrawler.
+    """
+    def setUp(self):
+        self.feed_list = rss_conf.links.keys()
+        self.props = rss_conf.properties
 
-def test_github():
-    # Github test
-    # TODO prevent from fake username
-    username = input("[+] GitHub Username: ")
-    github_result = GithubCrawler(username)
-    print(json.dumps(github_result, indent=4))
+    def test_links_are_available(self):
+        """
+        Test for news RSS feed links.
+        This test passes when all rss feed links are available.
+        """
+        for newspaper in self.feed_list:
+            response = requests.get(rss_conf.links[newspaper], headers=config.req_header)
+            response.raise_for_status()
+    
+    def test_rss_crawl(self):
+        """
+        Test for crawling rss feeds.
+        This test passes when all feed returns one or more articles.
+        """
+        for newspaper in self.feed_list:
+            rss_result = RSSCrawler(newspaper)
```

Comment:
Is RSSCrawler actually running the crawler? Unit tests shouldn't be making real RPC calls. The test we should be writing is:

- given a fake response from crawler X
- set up the expected parsed result Y
- ensure the crawler response is correctly parsed into Y, after running the parser.

Reply:
I need to change both of `RSSCrawler` and test code. I'll refactor it on next PR.